plugins/plugin-core-event-listener-state-storage/lib/StateStoragePlugin.ts

```diff
@@ -0,0 +1,177 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { existsSync, mkdirSync, writeFileSync } from "node:fs";
+import * as path from "node:path";
+
+import { Events, RootContext, Target } from "@syntest/analysis";
+import { ControlFlowProgram, format } from "@syntest/cfg";
+import { EventListenerPlugin } from "@syntest/module";
+import TypedEventEmitter from "typed-emitter";
+import Yargs = require("yargs");
+
+export type StateStorageOptions = {
+  stateStorageDirectory: string;
+};
+
+/**
+ * This graphing plugin creates a listener that creates an SVG based on the generated CFG.
+ *
+ * @author Dimitri Stallenberg
+ */
+export class StateStoragePlugin extends EventListenerPlugin {
+  constructor() {
+    super("state-storage", "Stores the state of the program");
+  }
+
+  setupEventListener(): void {
+    (<TypedEventEmitter<Events>>process).on(
+      "controlFlowGraphResolvingComplete",
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      <S>(
+        rootContext: RootContext<S>,
+        filePath: string,
+        cfp: ControlFlowProgram
+      ) => this.controlFlowGraphResolvingComplete(rootContext, filePath, cfp)
+    );
+    (<TypedEventEmitter<Events>>process).on(
+      "abstractSyntaxTreeResolvingComplete",
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      <S>(rootContext: RootContext<S>, filePath: string, ast: S) =>
+        this.abstractSyntaxTreeResolvingComplete(rootContext, filePath, ast)
+    );
+    (<TypedEventEmitter<Events>>process).on(
+      "targetExtractionComplete",
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      <S>(rootContext: RootContext<S>, filePath: string, target: Target) =>
+        this.targetExtractionComplete(rootContext, filePath, target)
+    );
+    (<TypedEventEmitter<Events>>process).on(
+      "dependencyResolvingComplete",
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      <S>(
+        rootContext: RootContext<S>,
+        filePath: string,
+        dependencies: string[]
+      ) => this.dependencyResolvingComplete(rootContext, filePath, dependencies)
+    );
+  }
+
+  override getOptions(
+    tool: string,
+    labels: string[],
+    command: string
+  ): Map<string, Yargs.Options> {
+    const optionsMap = new Map<string, Yargs.Options>();
+
+    if (!labels.includes("testing")) {
+      return optionsMap;
+    }
+
+    if (command !== "test") {
+      return optionsMap;
+    }
+
+    optionsMap.set("directory", {
```

Comment:
This name might be a bit generic. What about state-directory? What did we do for the other directories? We should have a common format

Reply:
because this is a plugin it will be prepended by the name of the plugin so it becomes state-storage-directory automatically.